design-documents/graph-ql/coverage/returns.graphqls

```diff
@@ -0,0 +1,197 @@
+    total_count: Int @doc(description: "Total count of customer returns")
+}
+
+type Return @doc(description: "Customer return") {
+    id: ID! @doc(description: "Return number")
+    order_id: String! @doc(description: "ID of the order based on which the return was created.")
+    creation_date: String! @doc(description: "The date when the return was requested.")
+    customer_email: String! @doc(description: "Email of the customer who created the return.")
+    customer_name: String @doc(description: "The name of the person returning the item")
+    status: ReturnStatus @doc(description: "Return status.")
+    shipping: ReturnShipping @doc(description: "Shipping information for the return.")
+    comments: [ReturnComment] @doc(description: "A list of comments posted for the return.")
+    items: [ReturnItem] @doc(description: "A list of items being returned.")
+    available_shipping_carriers: [ReturnShippingCarrier] @doc(description: "A list of shipping carriers available for returns.")
+}
+
+type ReturnItem {
+    id: ID!
+    product: ProductInterface!
+    custom_attributes: [CustomAttribute]
+    request_quantity: Float!
+    quantity: Float!
+    status: ReturnItemStatus!
+}
+
+# See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/custom-attributes-container.md
+type CustomAttribute {
+    id: ID!
+    label: String!
+    value: String! @doc(description: "JSON encoded value of the attribute.")
+}
+
+type ReturnComment {
+    id: ID! @doc(description: "Comment ID.")
+    created_by: String! @doc(description: "The name of the user who created posted the comment.")
+    created_at: String! @doc(description: "The date and time when the comment was posted.")
+    text: String! @doc(description: "The comment text.")
+}
+
+type ReturnShipping {
+    address: ReturnShippingAddress @doc(description: "Return shipping address, which is specified by the admin.")
+    tracking(id: ID): [ReturnShippingTracking] @doc(description: "Tracking information for all or a single tracking record when ID is provided.")
+}
+
+type ReturnShippingCarrier {
+    id: ID!
+    label: String!
+}
+
+type ReturnShippingTracking {
+    id: ID!
+    carier: ReturnShippingCarrier!
+    tracking_number: String!
+    status_type: ReturnShippingTrackingStatusType!
+    status: String!
+}
+
+enum ReturnShippingTrackingStatusType {
+    INFORMATION
+    ERROR
+}
+
+type ReturnShippingAddress {
+    contact_name: String
+    street: [String]!
+    city: String!
+    region: Region!
+    postcode: String!
+    country: Country!
+    telephone: String
+}
+
+enum ReturnStatus {
+    PENDING
+    AUTHORIZED
+    PARTIALLY_AUTHORIZED
+    RECEIVED
+    PARTIALLY_RECEIVED
+    APPROVED
+    PARTIALLY_APPROVED
+    REJECTED
+    PARTIALLY_REJECTED
+    DENIED
+    PROCESSED_AND_CLOSED
+    CLOSED
+}
+
+enum ReturnItemStatus {
+    PENDING
+    AUTHORIZED
+    RECEIVED
+    APPROVED
+    REJECTED
+    DENIED
+}
+
+type StoreConfig {
```

Comment:
Should we also be exposing the `sales/magento_rma/enabled_on_product` config? If my understanding is correct (very well could be wrong), that settings let's you open a return for a whole order, but not by line item.

https://github.com/magento/magento2ee/blob/2f1f767fbaeb4b9abc0cbad7fecbc1f070b2ce7f/app/code/Magento/Rma/Helper/Data.php#L637-L642

Reply:
I don't think this setting should be exposed directly on the storefront. My idea was that it should affect `items_eligible_for_return`